helpers/xspec_config.py

```diff
@@ -25,6 +26,58 @@ def clean(xs):
     "Remove all '' entries from xs, returning the new list."
     return [x for x in xs if x != '']
 
+# Try to support "optional" models; that is, those models
+# that are added (or deleted) in a version. At present
+# (XSPEC 12.9.0) this is only the nlapec model, so special
+# case it rather than come up with a generic method.
+#
+# The current approach to determining if the nlapec model is available
+# is to do a quick query on the model.dat file. However, this is
+# not specified in the build configuration - the best guess we have
+# here are:
+#  1) 'xspec-lib-dirs' + '../../spectral/manager/model.dat'
+#  2) $HEADAS + '../spectral/manager/model.dat'
+#
+# For now, the former approach is taken. This is not as flexible,
+# but requires less changes to the current build (since, at present,
+# there is no explicit requirement that the HEADAS environment variable
+# is set up for the build - although perhaps there should be).
+#
+
+def does_nlapec_model_exist(libdir):
+    """Can we find nlapec model in model.dat?
+
+    Parameters
+    ----------
+    libdir : str
+        The location of the XSPEC libraries. The model.dat file is
+        assumed to be located relative to this directory:
+        ../spectral/manager/model.dat.
+
+    Returns
+    -------
+    flag : bool
+        True if model.dat can be found and nlapec is found in it,
+        False otherwise.
+    """
+
+    fname = os.path.join(libdir, '../../spectral/manager/model.dat')
```

Comment:
No matter what I did I couldn't get the tests to pass... it took me a while to find the culprit: this line is hardly portable... in my setup, where I was using the libraries from `/soft/ciao/ots/lib`, the `fname` was a non-existent path, so the macro was not being set, so the model function was not built.

Reply:
Perhaps this location should be a config option in the `xspec` section of `setup.cfg`. 

Or we use the HEADAS environment variable and say it's `$HEADAS/../manager/model.dat`, but then this fails on CIAO builds, where it appears to be `$HEADAS/manager/model.dat` (or `$CIAO_HEADAS/...`). 

So the configuration option sounds best (from my not-really-looing-too-closely approach). Note that we don't necessarily want the path to the model.dat file, but to the directory containing the `model.dat` file, as we may one day want to also parse `mixmodel.dat`

Reply:
Note that the model is not built, but then in the tests the condition for assuming nlapec is not there is to check the version of xspec, which is why the tests fail.